osu.Game/Overlays/MusicController.cs

```diff
@@ -0,0 +1,307 @@
+                            Anchor = Anchor.Centre
+                        }
+                    }
+                },
+                progress = new DragBar
+                {
+                    Origin = Anchor.BottomCentre,
+                    Anchor = Anchor.BottomCentre,
+                    Height = 10,
+                    Colour = Color4.Orange,
+                    SeekRequested = seek
+                }
+            };
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+            if (current?.Track == null) return;
+
+            progress.UpdatePosition((float)(current.Track.CurrentTime / current.Track.Length));
+            playButton.Icon = current.Track.IsRunning ? FontAwesome.pause : FontAwesome.play_circle_o;
+
+            if (current.Track.HasCompleted) next();
+        }
+
+        private void workingChanged(object sender = null, EventArgs e = null)
+        {
+            if (beatmapSource.Value == current) return;
+            current = beatmapSource.Value;
+            updateCurrent(current, null);
+        }
+
+        private int findInPlaylist(Beatmap beatmap)
+        {
+            if (beatmap == null) return -1;
+            for (int i = 0; i < playList.Count; i++)
+                if (beatmap.BeatmapInfo.BeatmapSetID == playList[i].BeatmapSetID)
+                    return i;
+            return -1;
+        }
+
+        private void prev()
+        {
+            int i = findInPlaylist(current?.Beatmap);
+            if (i == -1)
+            {
+                if (playList.Count > 0)
+                    play(playList[0].Beatmaps[0], null);
+                else return;
+            }
+            i = (i - 1 + playList.Count) % playList.Count;
+            play(playList[i].Beatmaps[0], false);
+        }
+
+        private void next()
+        {
+            int i = findInPlaylist(current?.Beatmap);
+            if (i == -1)
+            {
+                if (playList.Count > 0)
+                    play(playList[0].Beatmaps[0], null);
+                else return;
+            }
+            i = (i + 1) % playList.Count;
+            play(playList[i].Beatmaps[0], true);
+        }
+
+        private void play(BeatmapInfo info, bool? isNext)
+        {
+            current = database.GetWorkingBeatmap(info, current);
+            trackManager.SetExclusive(current.Track);
+            current.Track.Start();
+            beatmapSource.Value = current;
+            updateCurrent(current, isNext);
+        }
+
+        private void updateCurrent(WorkingBeatmap beatmap, bool? isNext)
+        {
+            BeatmapMetadata metadata = beatmap.Beatmap.Metadata;
+            title.Text = metadata.TitleUnicode ?? metadata.Title;
+            artist.Text = metadata.ArtistUnicode ?? metadata.Artist;
+
+            Sprite newBackground;
+
+
+            newBackground = getScaledSprite(TextureLoader.FromStream(beatmap.Reader.ReadFile(metadata.BackgroundFile)));

```

Comment:
throws an exception if the file doesn't exist. need to gracefully handle this case.